### ISSUE TITLE
self-hosted gitlab not correctly detected in git_services

### ISSUE THREAD

**Issue by franklouwers** [NeogitOrg/neogit#1259]
### Description

I have a checked out git remote with a remote which is configured as:

```
origin	git@gitlab.x-plose.be:k8s/customer/k8s.git (fetch)
origin	git@gitlab.x-plose.be:k8s/customer/k8s.git (push)
```

The git remote is hosted on a self-hosted GitLab install at gitlab.x-plose.be.

I've added gitlab.x-plose.be as a git_service in my config with:

```lua
require("neogit").setup({
  git_services = {
    ["gitlab.x-plose.be"] =
    "https://gitlab.x-plose.be/${owner}/${repository}/merge_requests/new?merge_request[source_branch]=${branch_name}",
  },
})
```

The `b o` command to open a pull request reports it can't find the correct service.

### Neovim version

NVIM v0.10.0-dev-2883+g2c5fd3f96-Homebrew
Build type: Release
LuaJIT 2.1.1710088188

### Operating system and version

macOS 14.4.1

### Steps to reproduce

0. Be in a dir with a git repo with a remote `git@gitlab.x-plose.be:k8s/customer/k8s.git` in a branch
1. `nvim -nu minimal.lua`
2. Change something
3. `:Neogit`
4. Commit & Push
5. `b o` to open PR in a web browser

### Expected behavior

Web browser opens with the correct url.

### Actual behavior

No webbrowser opens, neovim shows `Pull request URL template not found for this branch's upstream`

### Minimal config

```Lua
-- NOTE: See the end of this file if you are reporting an issue, etc. Ignore all the "scary" functions up top, those are
-- used for setup and other operations.
local M = {}

local base_root_path = vim.fn.fnamemodify(debug.getinfo(1, "S").source:sub(2), ":p:h") .. "/.min"
function M.root(path)
  return base_root_path .. "/" .. (path or "")
end

function M.load_plugin(plugin_name, plugin_url)
  local package_root = M.root("plugins/")
  local install_destination = package_root .. plugin_name
  vim.opt.runtimepath:append(install_destination)

  if not vim.loop.fs_stat(package_root) then
    vim.fn.mkdir(package_root, "p")
  end

  if not vim.loop.fs_stat(install_destination) then
    print(string.format("> Downloading plugin '%s' to '%s'", plugin_name, install_destination))
    vim.fn.system({
      "git",
      "clone",
      "--depth=1",
      plugin_url,
      install_destination,
    })
    if vim.v.shell_error > 0 then
      error(string.format("> Failed to clone plugin: '%s' in '%s'!", plugin_name, install_destination),
        vim.log.levels.ERROR)
    end
  end
end

---@alias PluginName string The plugin name, will be used as part of the git clone destination
---@alias PluginUrl string The git url at which a plugin is located, can be a path. See https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols for details
---@alias MinPlugins table<PluginName, PluginUrl>

---Do the initial setup. Downloads plugins, ensures the minimal init does not pollute the filesystem by keeping
---everything self contained to the CWD of the minimal init file. Run prior to running tests, reproducing issues, etc.
---@param plugins? table<PluginName, PluginUrl>
function M.setup(plugins)
  vim.opt.packpath = {}                      -- Empty the package path so we use only the plugins specified
  vim.opt.runtimepath:append(M.root(".min")) -- Ensure the runtime detects the root min dir

  -- Install required plugins
  if plugins ~= nil then
    for plugin_name, plugin_url in pairs(plugins) do
      M.load_plugin(plugin_name, plugin_url)
    end
  end

  vim.env.XDG_CONFIG_HOME = M.root("xdg/config")
  vim.env.XDG_DATA_HOME = M.root("xdg/data")
  vim.env.XDG_STATE_HOME = M.root("xdg/state")
  vim.env.XDG_CACHE_HOME = M.root("xdg/cache")

  -- NOTE: Cleanup the xdg cache on exit so new runs of the minimal init doesn't share any previous state, e.g. shada
  vim.api.nvim_create_autocmd("VimLeave", {
    callback = function()
      vim.fn.system({
        "rm",
        "-r",
        "-f",
        M.root("xdg")
      })
    end
  })
end

-- NOTE: If you have additional plugins you need to install to reproduce your issue, include them in the plugins
-- table within the setup call below.
M.setup({
  plenary = "https://github.com/nvim-lua/plenary.nvim.git",
  telescope = "https://github.com/nvim-telescope/telescope.nvim",
  diffview = "https://github.com/sindrets/diffview.nvim",
  neogit = "https://github.com/NeogitOrg/neogit"
})
-- WARN: Do all plugin setup, test runs, reproductions, etc. AFTER calling setup with a list of plugins!
-- Basically, do all that stuff AFTER this line.
require("neogit").setup({
  git_services = {
    ["gitlab.x-plose.be"] =
    "https://gitlab.x-plose.be/${owner}/${repository}/merge_requests/new?merge_request[source_branch]=${branch_name}",
  },
}) -- For instance, setup Neogit
```

**Comment by CKolkey** [comment]
I believe the key needs to be `'git@gitlab.x-plose.be:k8s/customer/k8s.git'` and it will work.

**Comment by CKolkey** [comment]
Ahhh, I'm dumb. I didn't escape the pattern matching - your `-` is a valid lua pattern symbol and broke it. Easy fix - you should be set :)

**Comment by franklouwers** [comment]
Hi. I see you've closed this issue, but I don't see a commit / PR that mentions this. Is the fix in the Nightly? Or do I need to escape something in the config?

**Comment by CKolkey** [comment]
.... I was tired last night and forgot to push 😅 Sorry for the confusion, should be good now

**Comment by franklouwers** [comment]
Thanks @CKolkey I can confirm the url is detected now and my browser opens.

I still have another issue because of the way the gitlab urls are namespaced (git@host:group/subgroup/repo.git instead of the more traditional git@host:owner/repo.git), but I'll have some debugging time on my end ahead.

**Comment by CKolkey** [comment]
Here's where to start :) https://github.com/NeogitOrg/neogit/blob/nightly/lua/neogit/popups/branch/actions.lua#L266

**Comment by franklouwers** [comment]
Opened #1266 and #1267 to track and fix this
